compile/annotation.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package compile
+
+import (
+	"github.com/thriftrw/thriftrw-go/ast"
+)
+
+// AnnotationSpec describes and a thrift annotation
+type AnnotationSpec struct {
+	Name  string
+	Value string
+}
```

Comment:
I'm guessing your use case (and the majority use case for people using the
compile package) will involve looking annotations up by name. Maybe we should
use `type Annotations map[string]string` instead of `[]*AnnotationSpec`.

Although this may be prone to errors if we treat annotation names as case
insensitive because it's really easy to construct `Annotations{"Foo": "bar"}`
with keys that are not normalized. Further, if we normalize keys, we're now
losing the exact key that the user used (which may be something we want access
to).

How about keeping annotations case sensitive? This is compliant with what
Thrift already does. You'll just have to special case `compileAnnotations` to
return nil when `annotations` is empty.  For convenience, `type Annotations`
can have a `Get` function which handles the case where the map is nil.

```go
type Annotations map[string]string

func (a Annotations) Get(key string) (string, ok) {
  if a == nil {
    return "", false
  }
  ...
}
```